### PR TITLE
CI: Test and publish with Python 3.10

### PR DESCRIPTION
**Pull request by EwoutH**
CI: Test and publish with Python 3.10.

### CODE REVIEW

**Comment by EwoutH** [comment]
Please approve the 4 workflows waiting, to see if this PR passes CI.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -12,9 +12,9 @@ jobs:
     - uses: actions/checkout@v2
 
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
-        python-version: 3.8
+        python-version: 3.10
```
Comment (Conengmo) [inline .github/workflows/publish.yml:17]: @ocefpaf is it an issue to change the version number in the publish workflow? If we're unsure I'd suggest not changing this.

Comment (ocefpaf) [reply .github/workflows/publish.yml]: We can use "3.x" here.

File: .github/workflows/test_code.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
-        python-version: [3.6, 3.9]
+        python-version: [3.6, "3.10"]
```
Comment (ocefpaf) [inline .github/workflows/test_code.yml:17]: Let's quote all numbers for consistency.

File: .github/workflows/test_selenium.yml [hunk 3]
```diff
@@ -15,19 +15,19 @@ jobs:
...
-      uses: s-weigand/setup-conda@v1.0.5
+      uses: s-weigand/setup-conda@v1
       with:
         update-conda: true
 
     - name: Conda environment creation
       shell: bash -l {0}
       run: |
-        conda create --yes --name TEST python=3.9 pip --file requirements.txt --file requirements-dev.txt --channel conda-forge --strict-channel-priority
+        conda create --yes --name TEST python="3.10" pip --file requirements.txt --file requirements-dev.txt --channel conda-forge --strict-channel-priority
```
Comment (ocefpaf) [inline .github/workflows/test_selenium.yml:25]: This may break and we may need to wait b/c probably not all pkgs are available for 3.10 yet.

**Comment by ocefpaf** [comment]
Thanks @EwoutH!

**Pull request merged**